lua/lv-rust-tools/init.lua

```diff
@@ -0,0 +1,79 @@
+local opts = {
+    tools = { -- rust-tools options
+        -- automatically set inlay hints (type hints)
+        -- There is an issue due to which the hints are not applied on the first
+        -- opened file. For now, write to the file to trigger a reapplication of
+        -- the hints or just run :RustSetInlayHints.
+        -- default: true
+        autoSetHints = true,
+
+        -- whether to show hover actions inside the hover window
+        -- this overrides the default hover handler
+        -- default: true
+        hover_with_actions = true,
+
+        runnables = {
+            -- whether to use telescope for selection menu or not
+            -- default: true
+            use_telescope = true
+
+            -- rest of the opts are forwarded to telescope
+        },
+
+        inlay_hints = {
+            -- wheter to show parameter hints with the inlay hints or not
+            -- default: true
+            show_parameter_hints = true,
+
+            -- prefix for parameter hints
+            -- default: "<-"
+            parameter_hints_prefix = "<-",
+
+            -- prefix for all the other hints (type, chaining)
+            -- default: "=>"
+            other_hints_prefix  = "=>",
+
+            -- whether to align to the lenght of the longest line in the file
+            max_len_align = false,
+
+            -- padding from the left if max_len_align is true
+            max_len_align_padding = 1,
+
+            -- whether to align to the extreme right or not
+            right_align = false,
+
+            -- padding from the right if right_align is true
+            right_align_padding = 7,
+        },
+
+        hover_actions = {
+            -- the border that is used for the hover window
+            -- see vim.api.nvim_open_win()
+            border = {
+              {"╭", "FloatBorder"},
+              {"─", "FloatBorder"},
+              {"╮", "FloatBorder"},
+              {"│", "FloatBorder"},
+              {"╯", "FloatBorder"},
+              {"─", "FloatBorder"},
+              {"╰", "FloatBorder"},
+              {"│", "FloatBorder"}
+            },
+        }
+    },
+
+    -- all the opts to send to nvim-lspconfig
+    -- these override the defaults set by rust-tools.nvim
+    -- see https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md#rust_analyzer
+    server = {
+        cmd = {DATA_PATH .. "/lspinstall/rust/rust-analyzer"},
+        on_attach = require'lsp'.common_on_attach
+    }, -- rust-analyser options
+}
+
+require('rust-tools').setup(opts)
+
+vim.api.nvim_set_keymap("n", "<leader>lm", "<cmd>RustExpandMacro<CR>", {noremap = true, silent =  true})
```

Comment:
Might i suggest adding this, to make sure it only shows in rust files:
```lua
if vim.bo.filetype == 'rust' then
-- your keymappings
end
```

Reply:
I like the idea, thanks. This only has one major drawback which is that the keybindings will only show up if the first file you open is a rust file since nvim reads the whole configuration only once when it starts. If you just start the editor without giving it a file path and then open a rust file there won't be any keybindings. I guess we probably would have to fix that with an auto command.

Reply:
Yeah i agree, looking forward for this merge, thank you for your work :pray:

Reply:
Is this any good?
```lua
vim.api.nvim_exec(
    [[
    autocmd Filetype rust nnoremap <leader>lm <Cmd>RustExpandMacro<CR>
    autocmd Filetype rust nnoremap <leader>lH <Cmd>RustToggleInlayHints<CR>
    autocmd Filetype rust nnoremap <leader>le <Cmd>RustRunnables<CR>
    autocmd Filetype rust nnoremap <leader>lh <Cmd>RustHoverActions<CR>
    ]], true)

```

Reply:
Good job! I added that. Maybe at some point the project could introduce a more global solution to dynamically add and remove key bindings depending on the file type but for the time being this solution is probably the best we can do. Thanks!